model/client.go

```diff
@@ -518,6 +518,17 @@ func (c *Client) GetByUsername(username string, etag string) (*Result, *AppError
 	}
 }
 
+// getByEmail returns a user based on a provided username string. Must be authenticated.
+func (c *Client) GetByEmail(email string, etag string) (*Result, *AppError) {
```

Comment:
Hi @cpanato 

can you make this return (*User, *ResponseMetadata) as this function https://github.com/mattermost/platform/blob/master/model/client.go#L1356, we are trying to move away from the interface and being explicit on what the client actually returns.

thanks.